### PR TITLE
aws-sam-cli: 1.6.2 -> 1.11.0

### PR DESCRIPTION
**Pull request by pltanton**
Motivation of change: current version is really outdated, many features
missed. In my case there was lack of HttpApi support for
`sam local-run-api`.

Also aws-sam-translator: 1.27.0 -> 1.30.1 as dependency.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/aws-sam-translator/default.nix [hunk 1]
```diff
@@ -10,11 +10,11 @@
 
 buildPythonPackage rec {
   pname = "aws-sam-translator";
-  version = "1.27.0";
+  version = "1.30.1";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0m24cdiry8p74s5ysyriy6wks4ic8782sf21q4yvlwfn9hpi4n1j";
+    sha256 = "0d9ppd94x2kw404m49ajswmmxgdngbs4p5ajyrdvnlivfzqbv7dx";
```
Comment (jonringer) [inline pkgs/development/python-modules/aws-sam-translator/default.nix:17]: should should be a separate commit

Comment (pltanton) [reply pkgs/development/python-modules/aws-sam-translator/default.nix:17]: Sure. Done.

**Comment by pltanton** [comment]
What about this PR? Can anyone review it?

**Comment by pltanton** [comment]
@jonringer have I got any chance to merge it to upstream? :)

I hope it didn't sound rude (due to lack of my knowledge of English). But I can't find a recommended workflow for maintaining PRs for nixpkgs repository. Should I ping package maintainers or just wait for a several moth being sure, that it will be approved or declined someday?

**Comment by teto** [comment]
seems we have a newer version https://github.com/NixOS/nixpkgs/pull/104782

**Comment by pltanton** [comment]
Thanks, @teto. Closing this one then.
